tools/default.nix

```diff
@@ -0,0 +1,65 @@
+{ pkgs ? import <nixpkgs> {} , withEditor ? false }:
+
+pkgs.stdenv.mkDerivation rec {
+  name = "openclonk";
+
+  gitRef = pkgs.lib.commitIdFromGitRepo ../.git;
+
+  src = builtins.filterSource (path: type: ! builtins.elem (baseNameOf path) [
+    ".git" # leave out .git as it changes often in ways that do not affect the build
+    "default.nix" # default.nix might change, but the only thing that matters is what it evaluates to, and nix takes care of that
+    "result" # build result is irrelevant
+    "build"
+  ]) ./..;
+
+  enableParallelBuilding = true;
+
+  hardeningDisable = "format";
+
+  nativeBuildInputs = with pkgs; [ cmake ];
+
+  dontStrip = true;
+
+  buildInputs = with pkgs; [
+    SDL2 libvorbis libogg libjpeg libpng freetype glew tinyxml
+    openal freealut pkgconfig
```

Comment:
Good point on pkgconfig.

Those that you suggest are missing are clearly not, otherwise it wouldn't build ;)

- SDL_mixer: we're using OpenAL as recommended by @lluchs on IRC. It replaces SDL_mixer and apparently works better.
- zlib doesn't need to be specified explicitly — I think libpng brings it in
- OpenGL is brought in by GLEW (so we don't need to specify mesa, which has the additional benefit of not making it specific to platforms that use mesa, i.e. this might build on OSX too)
- if tinyxml isn't provided by the system, it will use a vendored version (in thirdparty/tinyxml) which increases the size of the build
- xrandr is brought in by something else, probably SDL2
- miniupnpc is an optional dependency. It doesn't build correctly when it is available, I'll need to look into this in more detail.